iosTraining-Reeen/WeatherViewController.swift

```diff
@@ -62,24 +62,59 @@ class ViewController: UIViewController {
         return stackView
     }()
     
+    private let weatherService: WeatherServiceProtocol
+    
+    init(weatherService: WeatherServiceProtocol) {
+        self.weatherService = weatherService
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         setupViews()
     }
 }
 
-private extension ViewController {
+private extension WeatherViewController {
+    func displayWeatherCondition() {
+        let conditionString = self.weatherService.getWeatherInformation()
+        weatherConditionImageView.image = imageForWeatherCondition(for: conditionString)
+    }
+    
+    func imageForWeatherCondition(for condition: String) -> UIImage? {
```

Comment:
説明不足だったかもですが、もうちょっと API は改善できそうです。
今は "for condition" というメッセージが重複してしまっているので、以下のようにしてはどうでしょう？

```suggestion
    func image(for condition: String) -> UIImage? {
```

こうすると、呼び出し側は

```swift
let conditionString = weatherService.getWeatherInformation()
let image = image(for: condition)
```

みたいになります。
どう呼び出し側に印象を与えたいかによって API は変わってきますが、他にも色々考えられそうです。

```swift
let conditionString = weatherService.getWeatherInformation()
let image1 = image(for: condition)
let image2 = weatherImage(for: condition)
let image3 = getImage(for: condition)
```

のような形です。

Reply:
関数名の修正忘れていました。
今後の実装によって変化するかもしれませんが、個人的には`getImage`という命名がとても好みだったので、これで修正させて頂きます！